### PR TITLE
Valero Conversion

### PR DESCRIPTION
**Pull request by h-mayorquin**
EDIT:

The full conversion pipeline for one session is here.

### CODE REVIEW

File: buzsaki_lab_to_nwb/valero/eventsinterface.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        if ripple_intervals.size == 0:
+            warnings.warn(f"No ripples found for session: {self.session_id}. Skipping ripple events interface")
+            return nwbfile
+
+        # Name and descriptions
+        field_in_mat_file_to_nwb_column_info = dict()
+        if "peaks" in ripples_data and ripples_data["peaks"].size != 0:
+            field_in_mat_file_to_nwb_column_info["peaks"] = dict(
+                name="peaks", description="Peak of the ripple.", data=ripples_data["peaks"]
+            )
```
Comment (CodyCBakerPhD) [inline buzsaki_lab_to_nwb/valero/eventsinterface.py:140]: From looking at `fCamk2_201012_sess1`, the data for `peaks` actually appears to be the _time_ at which the peak occurs, not the value of the peak (which I would have thought from the name)

As such, can we call this column `peak_time`? (and adjust description)

File: buzsaki_lab_to_nwb/valero/stimulilaserinterface.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        # Create the sites
+        site_description = "microscopic LED 10 x 15 µm each, 3 per shank. Each μLED has an emission area of 150 μm2"
+        location = "dorsal right hippocampus (antero-posterior 2.0 mm, mediolateral 1.5 mm, dorsoventral 0.6 mm)"
+
+        micro_led_ids_to_site = dict()
+        micro_led_ids = np.unique(pulse_micro_led)
+
+        for id in micro_led_ids:
+            optogenetic_site = OptogeneticStimulusSite(
+                name=f"Microled site in Neurolight probe with id {id}",
```
Comment (CodyCBakerPhD) [inline buzsaki_lab_to_nwb/valero/stimulilaserinterface.py:48]: I'm against the idea of including spaces in names as they ultimately result as keys in dictionaries (or worse, as attributes) when exploring an NWB file, and it's quite unusual as a practice to use spaces for dictionary keys

@h-mayorquin what do you think? Can we give these more typical object names, like `OptogeneticStimulusSite{id}`?

File: buzsaki_lab_to_nwb/valero/eventsinterface.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+        column_descriptions = dict(
+            peaks="TBD",
+            center="TBD",
+        )
+
+        column_data_dict = dict(
+            peaks=peaks,
+            center=center,
+        )
```
Comment (CodyCBakerPhD) [inline buzsaki_lab_to_nwb/valero/eventsinterface.py:94]: These had the same issue as the main ripple interface; these are definitely time values and so should end in `_time` at least

**Comment by CodyCBakerPhD** [comment]
Just some minor comments on a couple of fields, should be easy to fix

PSTH's for the one file with trials look very interesting

Happy that the ogen series widget works (can even see the trapezoidal pulse shape if you zoom in enough)

File: buzsaki_lab_to_nwb/valero/eventsinterface.py (hunk 1 above)
Comment (h-mayorquin) [reply buzsaki_lab_to_nwb/valero/eventsinterface.py:140]: Thanks for noticing that. I actually have peakAmpltidue as well so I was aware but the description and the convention of using _time is indeed missing.

File: buzsaki_lab_to_nwb/valero/stimulilaserinterface.py (hunk 2 above)
Comment (h-mayorquin) [reply buzsaki_lab_to_nwb/valero/stimulilaserinterface.py:48]: I think this makes sense. Yes. Changing it!

File: buzsaki_lab_to_nwb/valero/stimulilaserinterface.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            optogenetic_series_description = (
+                "μLEDs were controlled with current (2-4.5 μA generating 0.02-0.1μW of total light power;"
+                "ref (15)) provided by a 12-channel current generator (OSC1Lite, NeuroNex Michigan Hub)"
+                "driven by an Arduino, which delivered trapezoid (1ms rise time)"
+                "blue light (centered emission at 460 nm, emission surface area = 150 mm2) 20 ms pulses at"
+                "random sites with a randomly variable (40-60ms) offset"
+            )
+            optogenetic_site = micro_led_ids_to_site[id]
+            optogenetic_series = OptogeneticSeries(
+                name=f"Stimuli from microLED site {id}",
```
Comment (CodyCBakerPhD) [inline buzsaki_lab_to_nwb/valero/stimulilaserinterface.py:90]: Looks like this has the same problem with spaces

Comment (h-mayorquin) [reply buzsaki_lab_to_nwb/valero/stimulilaserinterface.py]: Thanks, Cody, corrected it.

File: buzsaki_lab_to_nwb/valero/stimulilaserinterface.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+            nwbfile.add_stimulus(optogenetic_series)
+
+
+# class ValeroLaserPulsesInterface(BaseDataInterface):
+#     def __init__(self, folder_path: FolderPathType):
+#         super().__init__(folder_path=folder_path)
+
+#     def run_conversion(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = False):
+#         self.session_path = Path(self.source_data["folder_path"])
```
[the remark this answers is not in the document]
Comment (h-mayorquin) [reply buzsaki_lab_to_nwb/valero/stimulilaserinterface.py]: Not anymore.

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
Will likely need further polishing as we get more feedback from the author and debug remaining data sources and run the full conversion, but this is a strong enough base

Feel free to merge when ready

**Pull request merged**